### PR TITLE
Add url to the SRPMBuildModel

### LINKED ISSUE #862: Link to SRPM file in the status page
Requested in https://github.com/packit/packit/issues/1010 by @pcahyna 

TODO:
- [ ] Get the SRPM URL from Copr.
- [ ] Add the SRPM link to the build page (e.g. https://prod.packit.dev/copr-build/69250)

### PR DESCRIPTION
**Pull request by mfocko**
Closes #862

- Add URL to the SRPMBuildModel that points to the built SRPM uploaded to
  COPR
- Set URL after starting COPR build
- Fix tests to reflect it

Signed-off-by: Matej Focko <mfocko@redhat.com>

TODO:
- [x] Update build_info
  - [x] Fix tests
- [x] alembic
  - [ ] Add links for older SRPMs
- [ ] Add link to requests to API (**?**)
- [x] Fix requre tests...

### CODE REVIEW

File: packit_service/models.py [hunk 1]
```diff
@@ -956,6 +956,7 @@ class SRPMBuildModel(Base):
     copr_builds = relationship("CoprBuildModel", back_populates="srpm_build")
     koji_builds = relationship("KojiBuildModel", back_populates="srpm_build")
     build_submitted_time = Column(DateTime, default=datetime.utcnow)
+    url = Column(Text)
```
Comment (lachmanfrantisek) [inline packit_service/models.py:959]: It's not so important, but would be nice if we can fill those values during the database upgrade for older SRPMs.

Comment (TomasTomecek) [reply packit_service/models.py:959]: oh my, that would be so cool

File: packit_service/worker/build/copr_build.py [hunk 2]
```diff
@@ -372,6 +372,7 @@ def run_build(
             build = self.api.copr_helper.copr_client.build_proxy.create_from_file(
                 ownername=owner, projectname=self.job_project, path=self.srpm_path
             )
+            self.srpm_model.set_url(build.source_package.url)
```
Comment (TomasTomecek) [inline packit_service/worker/build/copr_build.py:375]: nice, I'm so glad we don't need to construct the URL ourselves

File: packit_service/models.py (hunk 1 above)
Comment (lachmanfrantisek) [reply packit_service/models.py:959]: And not so hard I hope.

**Comment by mfocko** [comment]
> This issue has been marked as _stale_ because it hasn't seen any
> activity for the last 60 days.
> 
> _Stale_ issues are closed after 14 days, unless the label is removed
> by a maintainer or someone comments on it.
> 
> This is done in order to ensure that open issues are still relevant.
> 
> Thank you for your contribution! unicorn rocket robot
> 
> (Note: issues labeled with _pinned_ or _EPIC_ are
> never marked as stale.)

yikes... rebase is ready, will finish and push

File: packit_service/models.py (hunk 1 above)
Comment (mfocko) [reply packit_service/models.py:945]: with @lachmanfrantisek we agreed on not setting URLs for older SRPMs, my main reasons for not doing so is deletion of projects and related files after 60 days in COPR which would result in keeping 404 links in DB

We probably should also look into deletion of them after they're expired (looking at the `CoprBuildModel` there might be more expired URLs that we keep in database)

File: packit_service/service/templates/build_info.html [hunk 3]
```diff
@@ -11,7 +11,10 @@ <h1>
   <a
     href="{{ url_for('builds.get_srpm_build_logs_by_id', id_=build.srpm_build_id) }}"
   >
-    SRPM build
+    SRPM build {% if build.srpm_build.url %} (<a
+      href="{{ build.srpm_build.url }}"
+      >Download SRPM</a
+    >){% endif %}
```
Comment (lachmanfrantisek) [inline packit_service/service/templates/build_info.html:17]: What about not nesting the links?

```suggestion
    SRPM build</a>
    {% if build.srpm_build.url %} (<a
      href="{{ build.srpm_build.url }}"
      Download SRPM){% endif %}
```

(Note that I can't suggest an edit of the existing line so we need to add a right parenthesis after the `</a>` on the next line...)

**Review by lachmanfrantisek — APPROVED** [review]
Nice!

The API endpoint can be added later. Can you please create an issue for that?

**Pull request merged**